### PR TITLE
💚 DevOps -Add GitHub Actions to Build and Deploy to Azure Storage

### LINKED ISSUE #4: ✨Setup - Configure project settings
Cc: @adamcogan @pierssinclairssw @christianmorfordwaitessw 

To Myself
  
Cannot be completed until #4

### Tasks

- [x] Investigate required project settings
- [x] Add the projects settings to the project
- [x] Hook the Gatsby project to the Content repo

Thanks!

### LINKED ISSUE #12: 👷DevOps - Setup Staging pipeline (GitHub action)
Cc: @adamcogan @pierssinclairssw @christianmorfordwaitessw 

To Myself

### Tasks

- [x] Set up a GitHub action to deploy to Staging
- [x] Add any required Azure resources

Thanks!

### LINKED ISSUE #13: 👷DevOps - Setup Production pipeline (GitHub action)
Cc: @adamcogan @pierssinclairssw @christianmorfordwaitessw 

To Myself

### Tasks

- [x] Add any required Azure resources
- [x] Set up a GitHub action for Production according to the Staging pipeline

Thanks!

### LINKED ISSUE #18: 👷 DevOps - Setup Bicep template for Azure resources
Cc: @adamcogan @pierssinclairssw @christianmorfordwaitessw @brydeno 

To Myself

### Tasks

- [x] Work with Bryden / Piers to create a Bicep template for SSW Website

Thanks!

### PR DESCRIPTION
**Pull request by tkapa**
PR to create basic GitHub actions for building and deploying the Gatsby Site to Azure. 

These will require Environments/Secrets to be set up in the GitHub Repo.

#12 
#13 

Adding Workflows for Production and Staging:
- Scheduled build health check for `main` branch
- Staging and Production CI / Cd
    - Workflow 1: Deploy Resources to an Azure Resource Group
    - Workflow 2:
        - Build the Gatsby Website
        - Upload to blob

### CODE REVIEW

**Comment by AttackOnMorty** [comment]
- [ ] Please make the name of those 2 filenames consistent.
image

File: .github/workflows/staging-build-and-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# This is a basic workflow to help you get started with Actions
+
+name: Staging - Deploy Gatsby Site
+
+# Controls when the workflow will run
+on:
+ # TODO: Replace with on push to main
```
Comment (AttackOnMorty) [inline .github/workflows/staging-build-and-deploy.yml]: Those 3 workflows are now manually triggered. Shouldn't we change to `run after push to main` before merging this PR?

- build-main.yml
- staging-build-and-deploy.yml
- staging-deploy-azure-resources.yml

**Comment by tkapa** [comment]
> * [ ]  Please make the name of those 2 filenames consistent.
>   image

**Done ** - Files have been renamed to be consistent

**Review by AttackOnMorty — APPROVED** [review]
LGTM

**Pull request merged**